app/api/v2/resources/auth.py

```diff
@@ -0,0 +1,272 @@
+# app/api/resources/auth.py
+"""Contains user logic"""
+
+import os
+import sys
+import inspect
+import random
+import datetime
+import json
+
+from functools import wraps
+
+import jwt
+
+from flask import jsonify, request, make_response, session
+from flask_restful import Resource
+from flask_restful.reqparse import RequestParser
+from flasgger import swag_from
+from werkzeug.security import check_password_hash, generate_password_hash
+
+
+# solution to python 3 relative import messages
+# use the inspect module because for os.path.abspath(__file__),
+# the __file__ attribute is not always given
+auth_dir = os.path.dirname(os.path.abspath(
+    inspect.getfile(inspect.currentframe())))
+resources_dir = os.path.dirname(auth_dir)
+v2_dir = os.path.dirname(resources_dir)
+api_dir = os.path.dirname(v2_dir)
+app_dir = os.path.dirname(api_dir)
+sys.path.insert(0, app_dir)
+# sys.path.append(os.path.dirname)
+
+from app.db import db
+from app.models.blacklist import Blacklist
+from app.models.user import User
+from app.models.log import Log
+
+
+secret_key = os.environ.get('SECRET_KEY', 'MEGAtron35648')
+
+
+# RequestParser and added arguments will know which fields to accept and how to validate those
+user_request_parser = RequestParser(bundle_errors=True)
+
+user_request_parser.add_argument(
+    "first_name", type=str, required=True,
+    help="First name must be a valid string")
+
+user_request_parser.add_argument(
+    "last_name", type=str, required=True,
+    help="Last name must be a valid string")
+
+user_request_parser.add_argument(
+    "username", type=str, required=True,
+    help="Username must be a valid string")
+
+user_request_parser.add_argument(
+    "password", type=str, required=True,
+    help="Password is required")
+
+
+def string_empty(string_var):
+    """Return true if string is empty"""
+
+    return not isinstance(string_var, str) or string_var in [' ', '']
+
+
+def token_required(function):
+    @wraps(function)
+    def decorated_function(*args, **kwargs):
+        access_token = None
+
+        if "Authorization" in request.headers:
+            authorization = request.headers.get("Authorization", None)
+            if authorization:
+                access_token = authorization.split(" ")[1]
```

Comment:
`KeyError` if `authorization` is on long string i.e if `ABCD`, `authorization.split(" ")[1]` will be out of range.

Reply:
Noted. I will adjust it to
```
authorization = request.headers.get("Authorization", None)
if authorization:
    authorization_list = authorization.split(" ")
    if len(authorization_list) > 1:
        access_token = authorization_list[1]

    if not access_token:
        return {"message": "No token provided"}, 401
```